web/src/message_list.js

```diff
@@ -189,6 +189,11 @@ export class MessageList {
             !(
                 user_settings.web_mark_read_on_scroll_policy ===
                 web_mark_read_on_scroll_policy_values.never.code
+            ) &&
+            !(
+                user_settings.web_mark_read_on_scroll_policy ===
+                    web_mark_read_on_scroll_policy_values.conversation_only.code &&
+                !narrow_state.topic()
```

Comment:
This isn't the right check -- we want to check for conversation views. 

I think you want to look at something like `const filter = narrow_state.current_filter()`; and inspect `filter.sorted_term_types()` for whether it's `["stream", "topic"]` or `["dm"]`, with all the necessary checks for things being none so it doesn't throw an exception when looking at recent topics or whatnot.

Reply:
Hmm, this is the code for the topic function that I'm using to check for conversational view. 
```
export function topic() {
    if (current_filter === undefined) {
        return undefined;
    }
    const operands = current_filter.operands("topic");
    if (operands.length === 1) {
        return operands[0];
    }
    return undefined;
}
```
This has the checks for views that are none such as all message and draft. I don't really see much wrong with this but nonetheless, I will export `filter.is_covnersation_view` as we'll find ourselves using it elsewhere.

Reply:
The check was correct for whether there was a topic filter, but there could have been additional operands, like `stream:a topic:b search:c`.